js/components/thumbnail.js

```diff
@@ -4,15 +4,126 @@
  * @module Thumbnail
  */
 var React = require('react'),
-    Utils = require('./utils');
+    Utils = require('./utils'),
+    ReactDOM = require('react-dom');
 
 var Thumbnail = React.createClass({
+  getInitialState: function() {
+    this.positionY = 0;
+    this.positionX = -320;
+    this.imageWidth = 0;
+    this.imageHeight = 0;
+    this.thumbnailWidth = 80;
+    this.thumbnailHeight = 40;
+    return {};
+  },
+  componentDidMount: function() {
+    if (this.props.videoVr) {
+      this.setThumbnailSizes();
+      this.setImageSizes();
+      var yaw = this.props.vrViewingDirection.yaw;
+      var pitch = this.props.vrViewingDirection.pitch;
+      this.setCurrentViewVr(yaw, pitch);
+    }
+  },
+  componentWillReceiveProps: function(nextProps) {
+    if (this.props.vrViewingDirection !== nextProps.vrViewingDirection && this.props.videoVr) {
+      var yaw = nextProps.vrViewingDirection.yaw;
+      var pitch = nextProps.vrViewingDirection.pitch;
+      this.setCurrentViewVr(yaw, pitch);
+    }
+  },
   shouldComponentUpdate: function(nextProps) {
-    return (nextProps.hoverPosition != this.props.hoverPosition);
+    var updateHoverPositon = nextProps.hoverPosition != this.props.hoverPosition;
+    var updateFullscreen  = nextProps.fullscreen != this.props.fullscreen && this.props.videoVr;
+    var updateVrViewDIrection = nextProps.vrViewingDirection != this.props.vrViewingDirection;
```

Comment:
The I in direction is capitalized. It should be lower case.

Reply:
\+